wolfhsm/wh_utils.h

```diff
@@ -39,4 +40,50 @@ uint32_t wh_Utils_ntohl(uint32_t networklong);
 
 int wh_Utils_memeqzero(uint8_t* buffer, uint32_t size);
 
+/** Cache flushing and memory fencing synchronization primitives */
+/* Create a full sequential memory fence to ensure compiler memory ordering */
+#ifndef XMEMFENCE
+#define XMEMFENCE() __atomic_thread_fence(__ATOMIC_SEQ_CST)
```

Comment:
This is GCC specific - either needs to have a tree of compiler-specific #ifdefs or should be instead implemented by ports in `wolfhsm_config.h`?

Reply:
Agree.  This one is good for gcc's and clang and probably others too.  Move to settings.h